lib/preferences.js

```diff
@@ -0,0 +1,71 @@
+var fs = require('fs');
+var logs = require('./logs');
+var path = require('path');
+
+var preferenceFile = path.join(__dirname, '__preferences.json');
+
+var Preferences = function() {
+
+};
+
+Preferences.prototype.read = function(key, defaultValue) {
+  return this.readPreferences()
+    .then(function(contents) {
+      if (contents) {
+        return contents[key] || defaultValue;
+      } else {
+        return defaultValue;
+      }
+    })
+    .catch(function(error) {
+      logs.err('Error reading preference', key, error);
+      return defaultValue;
+    });
+};
+
+Preferences.prototype.write = function(key, value) {
+  // read existing preferences and write
+  var that = this;
+  var promise = new Promise(function(resolve, reject) {
+    that.readPreferences()
+      .then(function(contents) {
+        contents = contents || {};
+        contents[key] = value;
+        fs.writeFile(preferenceFile, JSON.stringify(contents), function(error) {
+          if (error) {
+            logs.err('Error writing preference', key, value);
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      })
+      .catch(function(error) {
+        reject(error);
+      });
+  });
+  return promise;
+};
+
+Preferences.prototype.readPreferences = function() {
+  var promise = new Promise(function(resolve, reject) {
+    fs.exists(preferenceFile, function(exists) {
+      if (exists) {
+        fs.readFile(preferenceFile, function(error, data) {
+          if (error) {
+            reject(error);
+          } else {
+            resolve(JSON.parse(data));
+          }
+        });
+      } else {
+        // we don't have any local preferences
+        // return falsy value
+        resolve();
+      }
+    });
+  });
+  return promise;
+};
+
+module.exports.Preferences = Preferences;
```

Comment:
Done. The reason why i did this originally was to avoid mix-casing on requires. 

```javascript
var path = require('path');
var Preferences = require('./preferences');
```
I thought this was looking weird and inconsistent.